frontend/src/components/cards/course-card.tsx

```diff
@@ -0,0 +1,42 @@
+import Link from "next/link";
+import { ratingToStars } from "~/lib/ratings-to-stars";
+import { type AvgMetrics, type CourseProfSchema } from "~/lib/zod-schema";
+import { Card, CardContent, CardHeader } from "../ui/card";
+
+export default async function CourseCard(props: {
+  courseId: number;
+  title: string;
+  professors: CourseProfSchema;
+  avgMetrics: AvgMetrics;
+}) {
+  console.log(props.professors);
+  return (
+    <Link href={`/courses/${props.courseId}`}>
+      <Card className="h-auto w-auto  ">
+        <CardHeader className="">
+          <div className="m-1 font-bold"> {props.title}</div>
+          <div className="m-1 text-gray-500"> {props.courseId}</div>
```

Comment:
I know it's not technically part of this issue, but this should be `props.courseCode`. `courseId` is for internal use (like directing to the proper course in the query), whereas `courseCode` will be the actual AU code (i.e. CPSC3000 or whatever). We can create a separate issue for this later if you want, no sweat to do it now